engine/class_modules/monk/sc_monk.cpp

```diff
@@ -4306,6 +4480,186 @@ struct yulon_spell_t : public monk_spell_t
+    target     = p;
+  }
+
+  double composite_aoe_multiplier(const action_state_t* state) const override
+  {
+    double cam = monk_heal_t::composite_aoe_multiplier( state );
+
+    if ( p()->buff.unity_within->check() )
+      cam *= 1 + p()->talent.conduit_of_the_celestials.unity_within_dmg_mult->effectN( 1 ).percent();
+
+    return cam;
+  }
+};
+
+struct celestial_conduit_t : public monk_spell_t
+{
+  struct celestial_conduit_dmg_t : public monk_spell_t
+  {
+    celestial_conduit_dmg_t( monk_t *p )
+      : monk_spell_t( p, "celestial_conduit_dmg", p->talent.conduit_of_the_celestials.celestial_conduit_dmg )
+    {
+      background = true;
+    }
+
+    double composite_aoe_multiplier( const action_state_t *state ) const override
+    {
+      double cam = monk_spell_t::composite_aoe_multiplier( state );
+
+      const std::vector<player_t *> &targets = state->action->target_list();
+
+      if ( !targets.empty() )
+        cam *= 1 + ( p()->talent.conduit_of_the_celestials.celestial_conduit->effectN( 1 ).percent() *
+                     std::min( (double)targets.size(),
+                               p()->talent.conduit_of_the_celestials.celestial_conduit->effectN( 3 ).base_value() ) );
+
+      return cam;
+    }
+  };
+
+  struct celestial_conduit_heal_t : public monk_heal_t
+  {
+    celestial_conduit_heal_t( monk_t *p )
+      : monk_heal_t( p, "celestial_conduit_heal", p->talent.conduit_of_the_celestials.celestial_conduit_heal )
+    {
+      background = true;
+      target     = p;
+    }
+
+    double composite_aoe_multiplier( const action_state_t *state ) const override
+    {
+      double cam = monk_heal_t::composite_aoe_multiplier( state );
+
+      const std::vector<player_t *> &targets = state->action->target_list();
+
+      if ( !targets.empty() )
+        cam *= 1 + ( p()->talent.conduit_of_the_celestials.celestial_conduit->effectN( 1 ).percent() *
+                     std::min( (double)targets.size(),
+                               p()->talent.conduit_of_the_celestials.celestial_conduit->effectN( 3 ).base_value() ) );
+
+      return cam;
+    }
+  };
+
+  celestial_conduit_dmg_t *damage;
+  celestial_conduit_heal_t *heal;
+
+  celestial_conduit_t( monk_t *p, util::string_view options_str )
+    : monk_spell_t( p, "celestial_conduit", p->talent.conduit_of_the_celestials.celestial_conduit ),
+      damage( new celestial_conduit_dmg_t( p ) ),
+      heal( new celestial_conduit_heal_t( p ) )
+  {
+    parse_options( options_str );
+
+    may_combo_strike = true;
+
+    tick_action = damage;
+
+    apply_affecting_aura( p->talent.conduit_of_the_celestials.flight_of_the_red_crane );
+    apply_affecting_aura( p->talent.conduit_of_the_celestials.jade_sanctuary );
+  }
+
+  bool usable_moving() const override
+  {
+    return true;
+  }
+
+  void execute() override
+  {
+    monk_spell_t::execute();
+
+    p()->buff.unity_within->trigger();
+  }
```

Comment:
No buff exists to carry the during-channel DR, movement speed modification, nor mechanic immunity.

Latter two are generally not modeled, though.